core/src/test/scala/org/broadinstitute/dsde/rawls/monitor/WorkspaceMigrationActorSpec.scala

```diff
@@ -0,0 +1,731 @@
+        migration.newGoogleProjectNumber shouldBe defined
+        migration.newGoogleProjectConfigured shouldBe defined
+      }
+    }
+
+
+  it should "error when there's an error on the workspace billing account" in
+    assertThrows[WorkspaceMigrationException] {
+      runMigrationTest {
+        for {
+          now <- nowTimestamp
+          workspace = spec.testData.v1Workspace.copy(
+            billingAccountErrorMessage = "oh noes :(".some,
+            name = UUID.randomUUID.toString,
+            workspaceId = UUID.randomUUID.toString
+          )
+
+          _ <- inTransaction { _ =>
+            createAndScheduleWorkspace(workspace) >> workspaceMigrations
+              .filter(_.workspaceId === workspace.workspaceIdAsUUID)
+              .map(_.started)
+              .update(now.some)
+          }
+
+          _ <- migrate
+
+        } yield fail("it did not fail")
+      }
+    }
+
+
+  it should "error when there's no billing account on the workspace" in
+    assertThrows[WorkspaceMigrationException] {
+      runMigrationTest {
+        for {
+          now <- nowTimestamp
+          workspace = spec.testData.v1Workspace.copy(
+            currentBillingAccountOnGoogleProject = None,
+            name = UUID.randomUUID.toString,
+            workspaceId = UUID.randomUUID.toString
+          )
+
+          _ <- inTransaction { _ =>
+            createAndScheduleWorkspace(workspace) >> workspaceMigrations
+              .filter(_.workspaceId === workspace.workspaceIdAsUUID)
+              .map(_.started)
+              .update(now.some)
+          }
+
+          _ <- migrate
+
+        } yield fail("it did not fail")
+      }
+    }
+
+
+  it should "error when the billing account on the billing project is invalid" in
+    assertThrows[WorkspaceMigrationException] {
+      runMigrationTest {
+        for {
+          now <- nowTimestamp
+
+          _ <- inTransaction { dataAccess =>
+            DBIO.seq(
+              createAndScheduleWorkspace(spec.testData.v1Workspace),
+              workspaceMigrations
+                .filter(_.workspaceId === spec.testData.v1Workspace.workspaceIdAsUUID)
+                .map(_.started)
+                .update(now.some),
+              dataAccess
+                .rawlsBillingProjectQuery
+                .filter(_.projectName === spec.testData.v1Workspace.namespace)
+                .map(_.invalidBillingAccount)
+                .update(true)
+            )
+          }
+
+          _ <- migrate
+
+        } yield fail("it did not fail")
+      }
+    }
+
+
+  it should "error when the billing account on the workspace does not match the billing account on the billing project" in
+    assertThrows[WorkspaceMigrationException] {
+      runMigrationTest {
+        for {
+          now <- nowTimestamp
+          workspace = spec.testData.v1Workspace.copy(
+            currentBillingAccountOnGoogleProject = RawlsBillingAccountName("greg").some,
```

Comment:
To ensure that this is different, could we base it on the other billing account? (maybe something like `spec.testData.v1Workspacec.billingAccount + "A"`?)

Just because nothing stops the billing account on the billing project from being `"greg"`

Reply:
Sure!